flexmeasures/ui/templates/base.html

```diff
@@ -358,6 +358,57 @@
         }
     });
 
+    /**
+     * Convert data from the new reference-based format
+     * back to the old format, for backward compatibility.
+     * 
+     * @param {Object} responseData - The data from the sensor data API.
+     * @returns {Object[]} - The adapted data in the old format.
+     */
+    function adaptChartData(responseData) {
+        // Check if it's the new reference-based format
+        if (responseData.data && responseData.sensors && responseData.sources) {
+            // Convert new format to old format
+            return responseData.data.map(point => {
+                const sensor = responseData.sensors[point.sid] || {};
+                const source = responseData.sources[point.src] || {};
+                
+                return {
+                    // Map shortened field names back to original names
+                    event_start: point.ts,
+                    event_value: point.val,
+                    belief_horizon: point.bh,
+                    scale_factor: point.sf,
+                    belief_time: point.bt,
+                    
+                    // Reconstruct full sensor and source objects
+                    sensor: {
+                        id: point.sid,
+                        name: sensor.name || '',
+                        sensor_unit: sensor.unit || '',
+                        unit: sensor.unit || '',
+                        description: sensor.description || '',
+                        asset_id: sensor.asset_id,
+                        asset_description: sensor.asset_description || ''
+                    },
+                    source: {
+                        id: point.src,
+                        name: source.name || '',
+                        model: source.model || '',
+                        type: source.type || 'other',
+                        description: source.description || ''
+                    },
+                    
+                    // Keep sensor_unit at root level for backward compatibility
+                    sensor_unit: sensor.unit || ''
```

Comment:
I want to ask @Ahmad-Wahid to make a plan to get rid of this.

We have the same info, just under `sensor.sensor_unit`. Maybe you can even load this PR and see if you can make your chart work. Or a small follow-up PR. (then this comment needs to become an issue)